src/CodeGen_LLVM.h

```diff
@@ -306,6 +306,15 @@ class CodeGen_LLVM : public IRVisitor {
      * different buffers */
     void add_tbaa_metadata(llvm::Instruction *inst, std::string buffer, Expr index);
 
+    /** Helpers for implementing fast integer division. */
+    // @{
+    // Compute high_half(a*b) >> shr
```

Comment:
I found the shift due to the high_half distinct from the shr confusing. Maybe clarify in the comment?